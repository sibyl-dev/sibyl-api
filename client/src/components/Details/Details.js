import React, { Component } from 'react';
import { connect } from 'react-redux';
import Select from 'react-select';
import Button from '@material-ui/core/Button';
import { TableFullIcon, TableSplitIcon, SortIcon, ChevronDownIcon, ChevronUpIcon } from '../../assets/icons/icons';
import DashWrapper from '../common/DashWrapper';
import Search from '../common/Search';
import { CategorySelect } from '../common/Form';
import Loader from '../common/Loader';
import MetTooltip from '../common/MetTooltip';
import { BiProgressBar } from '../common/ProgressBars';
import {
  sortFeaturesByContribAction,
  // setFilterValuesAction,
  setFilterCategsAction,
  setContribFiltersAction,
  // setFeatureTypeFilterAction,
  setFeatureTypeSortContribDirAction,
  setFeatureTypeFilterCategsAction,
  setFilterCriteriaAction,
} from '../../model/actions/features';
import { getIsEntitiesLoading, getCurrentEntityData, getIsEntityContribLoading } from '../../model/selectors/entities';

import {
  getIsFeaturesLoading,
  getFeaturesData,
  getIsCategoriesLoading,
  getFeatureCategories,
  getSortingContribDir,
  // getSelectedFilterValues,
  getFilterCategories,
  getCurrentContribFilters,
  // getFeatureTypeFilters,
  getFeatureTypeSortContribDir,
  getGrouppedFeatures,
  getFeatureTypeFilterCategs,
} from '../../model/selectors/features';
import { setUserActionRecording } from '../../model/actions/userActions';

import { setActivePageAction } from '../../model/actions/sidebar';
import './Details.scss';

// const filterValues = [
//   { value: 'all', label: 'All Values', isFixed: true },
//   { value: 'binary', label: 'True/False', isFixed: true },
//   { value: 'numeric', label: 'Numerical', isFixed: true },
// ];

const contribFilters = [
  { value: 'all', label: 'All Contributions', isFixed: true },
  { value: 'risk', label: 'Risk', isFixed: true },
  { value: 'protective', label: 'Protective', isFixed: true },
];

const initialContributionView = {
  isCombinedExpanded: false,
  isPositiveViewExpanded: false,
  isNegativeViewExpanded: false,
};

export class Details extends Component {
  constructor(props) {
    super(props);
    this.state = {
      viewMode: 'unified',
      featureContribView: initialContributionView,
    };
  }

  componentDidMount() {
    const userData = {
      element: 'details_page',
      action: 'click',
    };
    this.props.setUserActions(userData);
    this.props.setPageName('Details');
  }

  componentWillUnmount() {
    this.props.setContribFilters('all');
  }

  getExpanded(featureContribView, viewMode, featureType = null) {
    if (viewMode === 'unified') {
      return featureContribView.isCombinedExpanded;
    }

    if (featureType === 'positiveFeatures') return featureContribView.isPositiveViewExpanded;

    return featureContribView.isNegativeViewExpanded;
  }

  changeViewMode(viewMode) {
    this.setState(
      {
        viewMode,
      },
      () => this.recordUserAction(),
    );
  }

  renderSubheader() {
    const { viewMode, featureContribView } = this.state;
    const { isNegativeViewExpanded, isPositiveViewExpanded } = featureContribView;

    return (
      <div className="sub-header">
        <ul>
          <li className="search-field-holder">
            {viewMode === 'split' ? (
              <Search disabled={!isNegativeViewExpanded || !isPositiveViewExpanded} />
            ) : (
              <h4>Factor Contributions</h4>
            )}
          </li>
          {viewMode === 'split' ? (
            <li className={`expand-tip ${isNegativeViewExpanded && isPositiveViewExpanded ? 'hide' : ''}`}>
              <span>Click &ldquo;Show All Factors&ldquo; to enable Search and Filter in both tables</span>
            </li>
          ) : null}
          <li>
            <MetTooltip title="Single Table View" placement="top">
              <button
                type="button"
                className={`view-full ${viewMode === 'unified' ? 'active' : ''}`}
                onClick={() => this.changeViewMode('unified')}
              >
                <TableFullIcon />
              </button>
            </MetTooltip>
            <MetTooltip title="Split View Comparison" placement="top">
              <button
                type="button"
                className={`view-split ${viewMode === 'split' ? 'active' : ''}`}
                onClick={() => this.changeViewMode('split')}
              >
                <TableSplitIcon />
              </button>
            </MetTooltip>
          </li>
        </ul>
      </div>
    );
  }

  renderDashHeader(featureType) {
    const { viewMode, featureContribView } = this.state;
    const { isCombinedExpanded } = featureContribView;
    const {
      // setFilterValues,
      // currentFilterValue,
      isCategoriesLoading,
      featureCategories,
      setFilterCategories,
      currentFilterCategs,
      setContribFilters,
      currentContribFilters,
      // setFeatureFilters,
      // featureTypeFilters,
      setFeatureTypeFilterCategs,
      currentFeatureTypeCategs,
    } = this.props;

    // const setFeatureFilterValues = (filterValue) =>
    //   featureType !== 'all' ? setFeatureFilters(featureType, filterValue) : setFilterValues(filterValue);

    // const getFilterValue = () =>
    //   featureType !== 'all'
    //     ? filterValues.filter((currentValue) => currentValue.value === featureTypeFilters[featureType])
    //     : filterValues.filter((currentValue) => currentValue.value === currentFilterValue);

    const setFeatureCategsFilter = (categs) =>
      featureType !== 'all' ? setFeatureTypeFilterCategs(featureType, categs) : setFilterCategories(categs);

    const getCurrentCategs = () =>
      featureType !== 'all' ? currentFeatureTypeCategs[featureType] : currentFilterCategs;

    return (
      !isCategoriesLoading && (
        <header className="dash-header">
          <ul className="dash-controls">
            {viewMode === 'unified' && (
              <>
                <li>
                  <Search disabled={!isCombinedExpanded} />
                </li>
                <li className="sep" />
              </>
            )}
            <li className={`category-select-${featureType}`}>
              <CategorySelect
                options={featureCategories}
                onChange={(selectedCategories) => setFeatureCategsFilter(selectedCategories)}
                value={getCurrentCategs()}
                disabled={!this.getExpanded(featureContribView, viewMode, featureType)}
              />
            </li>
            {/* <li>
              <Select
                isSearchable={false}
                isMulti={false}
                classNamePrefix="sibyl-select"
                className="sibyl-select"
                options={filterValues}
                placeholder="All Values"
                value={getFilterValue()}
                onChange={(filterValue) => setFeatureFilterValues(filterValue.value)}
              />
            </li> */}
            {viewMode === 'unified' && (
              <>
                <li>
                  <Select
                    isSearchable={false}
                    isMulti={false}
                    classNamePrefix="sibyl-select"
                    className="sibyl-select"
                    options={contribFilters}
                    isDisabled={!this.getExpanded(featureContribView, viewMode)}
                    placeholder="Contribution"
                    onChange={(currentFilters) => setContribFilters(currentFilters.value)}
                    value={contribFilters.filter((currentContrib) => currentContrib.value === currentContribFilters)}
                  />
                </li>
                <li className={`expand-tip ${isCombinedExpanded ? 'hide' : ''}`}>
                  <span>Click &ldquo;Show All Factors&ldquo; to enable Search and Filter</span>
                </li>
              </>
            )}
          </ul>
        </header>
      )
    );
  }

  toggleDash(featureContribView, viewMode, featureType) {
    const { setContribFilters, setFilterCriteria, setFeatureTypeFilterCategs, setFilterCategories } = this.props;
    const { isCombinedExpanded, isPositiveViewExpanded, isNegativeViewExpanded } = featureContribView;

    // reset filters
    if (featureContribView) {
      setContribFilters('all');
      setFilterCriteria('');

      if (featureType !== 'all') {
        setFeatureTypeFilterCategs(featureType, null);
      } else {
        setFilterCategories(null);
      }
    }

    // create new state for featureContribView
    const updatedFeatureContribView = {
      ...featureContribView,
      isCombinedExpanded: viewMode === 'unified' ? !isCombinedExpanded : isCombinedExpanded,
    };

    if (viewMode !== 'unified') {
      if (featureType === 'positiveFeatures') {
        updatedFeatureContribView.isPositiveViewExpanded = !isPositiveViewExpanded;
      } else {
        updatedFeatureContribView.isNegativeViewExpanded = !isNegativeViewExpanded;
      }
    }

    this.setState(
      {
        viewMode,
        featureContribView: updatedFeatureContribView,
      },
      () => this.recordUserAction,
    );
  }

  renderFooter(featureType, isDataLoading) {
    const { isCategoriesLoading, grouppedFeatures, features } = this.props;
    const { positiveFeaturesContrib, negativeFeaturesContrib } = grouppedFeatures;
    const { processedFeatures } = features;

    const { featureContribView, viewMode } = this.state;
    const featureContribViewValue = this.getExpanded(featureContribView, viewMode, featureType);

    const getResultsCount = () => {
      if (isDataLoading) {
        return 0;
      }

      const limitNumber = viewMode === 'unified' ? 10 : 5;
      const positiveLength =
        positiveFeaturesContrib.length > limitNumber && !featureContribViewValue
          ? limitNumber
          : positiveFeaturesContrib.length;
      const negativeLength =
        negativeFeaturesContrib.length > limitNumber && !featureContribViewValue
          ? limitNumber
          : negativeFeaturesContrib.length;
      const processedLength =
        processedFeatures.length > limitNumber && !featureContribViewValue ? limitNumber : processedFeatures.length;

      return featureType !== 'all'
        ? featureType === 'positiveFeatures'
          ? positiveLength
          : negativeLength
        : processedLength;
    };

    const toggleButton = () => (
      <Button
        className="expand-button"
        onClick={() => this.toggleDash(featureContribView, viewMode, featureType)}
        startIcon={featureContribViewValue ? <ChevronUpIcon /> : <ChevronDownIcon />}
      >
        {' '}
        {featureContribViewValue ? 'HIDE EXTRA FACTORS' : 'SHOW ALL FACTORS'}
      </Button>
    );

    return (
      !isCategoriesLoading && (
        <div className="dash-footer">
          <p>
            Showing <span>{getResultsCount()}</span> factors
          </p>
          {toggleButton()}
        </div>
      )
    );
  }

  // getting the contribution max value to set the min/max range (-range, range)
  getContributionsMaxValue() {
    let maxRange = 0;
    const { processedFeatures } = this.props.features;

    processedFeatures.map((currentFeature) => {
      const { contributionValue } = currentFeature;
      maxRange = maxRange > Math.abs(contributionValue) ? maxRange : Math.abs(contributionValue);
      return null;
    });
    return maxRange;
  }

  getFeatureType(feature) {
    const { entityData } = this.props;
    const { name, type } = feature;
    return type === 'binary' ? (entityData.features[name] > 0 ? 'True' : 'False') : entityData.features[name];
  }

  getFeatureCategoryColor(feature) {
    const { featureCategories } = this.props;
    const colorIndex = featureCategories.findIndex((currentCategory) => currentCategory.name === feature);

    if (colorIndex === -1) {
      return null;
    }

    return (
      <>
        <MetTooltip title={featureCategories[colorIndex].name} placement="top">
          <i className="bullet" style={{ background: featureCategories[colorIndex].color }} />
        </MetTooltip>
        <span className="feature-category-label">{featureCategories[colorIndex].abbreviation}</span>
      </>
    );
  }

  setSortContribDirection() {
    const { setSortContribDir, currentSortDir } = this.props;
    const currentDirection = currentSortDir === 'asc' ? 'desc' : 'asc';
    setSortContribDir(currentDirection);
  }

  setFeatureSortContribDir(featureType) {
    const { setFeatureSortDir, currentFeatureTypeSortDir } = this.props;
    const direction = currentFeatureTypeSortDir[featureType] === 'asc' ? 'desc' : 'asc';
    setFeatureSortDir(featureType, direction);
  }

  renderUnifiedMode() {
    const { isEntityLoading, isFeaturesLoading, isCategoriesLoading, isEntityContribLoading, features } = this.props;
    const { processedFeatures } = features;

    const isDataLoading = isEntityLoading || isFeaturesLoading || isCategoriesLoading || isEntityContribLoading;
    return (
      <div>
        {this.renderDashHeader('all', isDataLoading)}
        {this.renderFeatures(processedFeatures, isDataLoading)}
        {this.renderFooter('all', isDataLoading)}
      </div>
    );
  }

  renderFeatures(features, isDataLoading, featuresType = 'all') {
    const maxContributionRange = !isDataLoading ? this.getContributionsMaxValue() : 0;
    const { viewMode, featureContribView } = this.state;
    const featureContribViewValue = this.getExpanded(featureContribView, viewMode, featuresType);

    const featuresToShow = !featureContribViewValue && features ? features.slice(0, 10) : features;

    return (
      <div className="sticky-wrapper scroll-style">
        <table className="dash-table sticky-header">
          <thead>
            <tr>
              <th className="align-center" width="10%">
                Category
              </th>
              <th>Factor</th>
              <th className="align-right">Value</th>
              <th className="align-center" width="15%">
                <ul className="sort">
                  <li>Contribution</li>
                  <li>
                    <button
                      type="button"
                      onClick={() =>
                        viewMode === 'unified'
                          ? this.setSortContribDirection()
                          : this.setFeatureSortContribDir(featuresType)
                      }
                    >
                      <SortIcon />
                    </button>
                  </li>
                </ul>
              </th>
            </tr>
          </thead>
          <tbody>
            <Loader isLoading={isDataLoading}>
              {featuresToShow && featuresToShow.length > 0 ? (
                featuresToShow.map((currentFeature) => (
                  <tr key={currentFeature.name}>
                    <td className="align-center">{this.getFeatureCategoryColor(currentFeature.category)}</td>
                    <td>{currentFeature.description}</td>
                    <td className="align-right">{this.getFeatureType(currentFeature)}</td>
                    <td className="align-center" width="145">
                      <BiProgressBar
                        percentage={currentFeature.contributionValue}
                        width="110"
                        height="10"
                        maxRange={maxContributionRange}
                        isSingle={viewMode === 'split'}
                      />
                    </td>
                  </tr>
                ))
              ) : (
                <tr>
                  <td colSpan="4" className="align-center">
                    No Matches found...
                  </td>
                </tr>
              )}
            </Loader>
          </tbody>
        </table>
      </div>
    );
  }

  renderSplitMode() {
    const { isFeaturesLoading, isEntityContribLoading, grouppedFeatures } = this.props;
    const { positiveFeaturesContrib, negativeFeaturesContrib } = grouppedFeatures;
    const isDataLoading = isEntityContribLoading || isFeaturesLoading;

    return (
      <div className="split-wrapper">
        <div className="split-side">
          <h4>Risk Factors</h4>
          <div className="split-container">
            {this.renderDashHeader('positiveFeatures', isDataLoading)}
            {this.renderFeatures(positiveFeaturesContrib, isDataLoading, 'positiveFeatures')}
            {this.renderFooter('positiveFeatures', isDataLoading)}
          </div>
        </div>
        <div className="split-separator" />
        <div className="split-side">
          <h4>Protective Factors</h4>
          <div className="split-container">
            {this.renderDashHeader('negativeFeatures', isDataLoading)}
            {this.renderFeatures(negativeFeaturesContrib, isDataLoading, 'negativeFeatures')}
            {this.renderFooter('negativeFeatures', isDataLoading)}
          </div>
        </div>
      </div>
    );
  }

  recordUserAction() {
    const { viewMode } = this.state;
    const userData = {
      element: viewMode === 'split' ? 'split_view' : 'unified_view',
      action: 'click',
    };
    this.props.setUserActions(userData);
  }

  render() {
    const { viewMode } = this.state;

    return (
      <div>
        {this.renderSubheader()}
        <div className="component-wrapper no-shadow">
          <DashWrapper className={`${viewMode === 'split' ? 'no-shadow' : ''} `}>
            {viewMode === 'split' ? this.renderSplitMode() : this.renderUnifiedMode()}
          </DashWrapper>
        </div>
      </div>
    );
  }
}

export default connect(
  (state) => ({
    isEntityLoading: getIsEntitiesLoading(state),
    isFeaturesLoading: getIsFeaturesLoading(state),
    isCategoriesLoading: getIsCategoriesLoading(state),
    isEntityContribLoading: getIsEntityContribLoading(state),
    entityData: getCurrentEntityData(state),
    features: getFeaturesData(state),
    featureCategories: getFeatureCategories(state),
    currentSortDir: getSortingContribDir(state),
    // currentFilterValue: getSelectedFilterValues(state),
    currentFilterCategs: getFilterCategories(state),
    currentContribFilters: getCurrentContribFilters(state),
    // featureTypeFilters: getFeatureTypeFilters(state),
    currentFeatureTypeSortDir: getFeatureTypeSortContribDir(state),
    grouppedFeatures: getGrouppedFeatures(state),
    currentFeatureTypeCategs: getFeatureTypeFilterCategs(state),
  }),
  (dispatch) => ({
    setSortContribDir: (direction) => dispatch(sortFeaturesByContribAction(direction)),
    // setFilterValues: (filterValue) => dispatch(setFilterValuesAction(filterValue)),
    setFilterCategories: (filterCategs) => dispatch(setFilterCategsAction(filterCategs)),
    setFilterCriteria: (filterValue) => dispatch(setFilterCriteriaAction(filterValue)),

    setContribFilters: (currentContribFilters) => dispatch(setContribFiltersAction(currentContribFilters)),
    // setFeatureFilters: (featureType, filter) => dispatch(setFeatureTypeFilterAction(featureType, filter)),
    setFeatureSortDir: (featureType, direction) => dispatch(setFeatureTypeSortContribDirAction(featureType, direction)),
    setFeatureTypeFilterCategs: (featureType, categs) =>
      dispatch(setFeatureTypeFilterCategsAction(featureType, categs)),
    setUserActions: (userAction) => dispatch(setUserActionRecording(userAction)),
    setPageName: (pageName) => dispatch(setActivePageAction(pageName)),
  }),
)(Details);
