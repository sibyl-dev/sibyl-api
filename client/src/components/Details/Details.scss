.sub-header {
    h4 {
        margin: 10px 0;
    }
    .search-wrapper {
        max-width: 500px;
    }
}
.split-wrapper {
    display: flex;

    .split-separator {
        min-height: 100%;
        border-left: 1px dashed #bdbdbd;
        margin: 0 10px;
    }

    .split-container {
        flex: 1;
        box-shadow: 0px 2px 10px rgba(0, 0, 0, 0.2);
        padding: 7px 0 0 0;
        border-radius: 8px;
        min-height: 93%;
    }

    .split-side {
        width: 100%;
        h4 {
            margin-bottom: 24px;
        }
    }
}
