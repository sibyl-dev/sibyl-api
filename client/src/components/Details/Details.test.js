import React from 'react';
import { Details } from './Details';
import { renderWithStore, TestWrapper } from '../../tests/setup';

jest.mock('../common/Form', () => ({
  CategorySelect: () => 'Category select here',
}));

describe('Testing Details component -> ', () => {
  const store = {
    isEntityLoading: false,

    features: {
      processedFeatures: [
        {
          name: 'PRI_FOCUS_PLSM_PAST730_DUMMY',
          description: 'the child in focus had a child welfare placement in the last 730 days',
          type: 'binary',
          category: 'placement history',
          PRI_FOCUS_PLSM_PAST730_DUMMY: 1,
          contributionValue: 0.30542072380625584,
        },
        {
          name: 'PRI_CBMS_OTHC_CW_1_COUNT',
          description:
            'Count of other children in the referral (excl. child in focus) who received Colorado Works (CW) benefits in the 1 year prior to referral, excl. clients who are actively receiving the service.',
          type: 'binary',
          category: 'program involvement',
          PRI_CBMS_OTHC_CW_1_COUNT: 4,
          contributionValue: 0.24523142917914945,
        },
      ],
    },
    featureCategories: [
      { name: 'placement history', color: '#27ae60' },
      { name: 'program involvement', color: '#9e09b8' },
    ],
    entityData: {
      features: [
        {
          PRI_FOCUS_PLSM_PAST730_DUMMY: 1,
        },
      ],
    },
    grouppedFeatures: {
      negativeFeaturesContrib: {
        PRI_PRNT_AGE_2024_COUNT: 2,
        category: 'demographics',
        contributionValue: -0.43298241559380907,
        description: 'Counts of the number of parents that are 20<=age<25',
        name: 'PRI_PRNT_AGE_2024_COUNT',
        type: 'numeric',
      },
      positiveFeaturesContrib: {
        PRI_FOCUS_AGE_TOD: 1,
        category: 'demographics',
        contributionValue: 0.17053285540215513,
        description: 'Child in focus is between the ages of 1 and 3 at time of referral',
        name: 'PRI_FOCUS_AGE_TOD',
        type: 'binary',
      },
    },
  };
  it('Should render without crashing', () => {
    const detailsComponent = renderWithStore(
      {},
      <TestWrapper>
        <Details {...store} />
      </TestWrapper>,
    );

    expect(detailsComponent).toMatchSnapshot();
  });
});
