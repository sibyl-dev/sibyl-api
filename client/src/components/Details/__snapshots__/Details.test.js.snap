// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Details component ->  Should render without crashing (unified mode) 1`] = `
<div>
  <div
    class="sub-header"
  >
    <ul>
      <li>
        <h4>
          Risk Factors List
        </h4>
      </li>
      <li>
        <button
          class="view-full active"
          disabled=""
          type="button"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 24 16"
            width="24"
          >
            <path
              d="M0 2C0 0.895431 0.895431 0 2 0H22C23.1046 0 24 0.895431 24 2V14C24 15.1046 23.1046 16 22 16H2C0.89543 16 0 15.1046 0 14V2Z"
              fill="#E0E0E0"
              opacity="0.7"
            />
          </svg>
        </button>
        <button
          class="view-split "
          type="button"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 28 16"
            width="28"
          >
            <path
              d="M0 2C0 0.895431 0.895431 0 2 0H9C10.1046 0 11 0.895431 11 2V14C11 15.1046 10.1046 16 9 16H2C0.895431 16 0 15.1046 0 14V2Z"
              fill="#E0E0E0"
            />
            <path
              d="M13 1C13 0.447715 13.4477 0 14 0C14.5523 0 15 0.447715 15 1V15C15 15.5523 14.5523 16 14 16C13.4477 16 13 15.5523 13 15V1Z"
              fill="#E0E0E0"
            />
            <path
              d="M17 2C17 0.895431 17.8954 0 19 0H26C27.1046 0 28 0.895431 28 2V14C28 15.1046 27.1046 16 26 16H19C17.8954 16 17 15.1046 17 14V2Z"
              fill="#E0E0E0"
            />
          </svg>
        </button>
      </li>
    </ul>
  </div>
  <div
    class="component-wrapper no-shadow"
  >
    <div
      class="dash-wrapper  "
    >
      <div
        class="dash-body"
      >
        <div>
          <header
            class="dash-header"
          >
            <ul
              class="dash-controls"
            >
              <li>
                <div
                  class="search-wrapper"
                >
                  <div
                    aria-expanded="false"
                    class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
                    role="combobox"
                  >
                    <div>
                      <input
                        aria-autocomplete="list"
                        autocapitalize="none"
                        autocomplete="off"
                        class="MuiAutocomplete-input MuiAutocomplete-inputFocused"
                        id="highlights-demo"
                        placeholder="Search feature"
                        spellcheck="false"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </li>
              <li
                class="sep"
              />
              <li>
                <div
                  class="sibyl-select css-2b097c-container"
                >
                  <div
                    class="sibyl-select__control css-yk16xz-control"
                  >
                    <div
                      class="sibyl-select__value-container sibyl-select__value-container--is-multi css-g1d714-ValueContainer"
                    >
                      <div
                        class="sibyl-select__placeholder css-1wa3eu0-placeholder"
                      >
                        Category
                      </div>
                      <input
                        aria-autocomplete="list"
                        class="css-62g3xt-dummyInput"
                        id="react-select-2-input"
                        readonly=""
                        tabindex="0"
                        value=""
                      />
                    </div>
                    <div
                      class="sibyl-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="sibyl-select__indicator-separator css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="sibyl-select__indicator sibyl-select__dropdown-indicator css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-6q0nyr-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </li>
              <li>
                <div
                  class="sibyl-select css-2b097c-container"
                >
                  <div
                    class="sibyl-select__control css-yk16xz-control"
                  >
                    <div
                      class="sibyl-select__value-container css-g1d714-ValueContainer"
                    >
                      <div
                        class="sibyl-select__placeholder css-1wa3eu0-placeholder"
                      >
                        All Values
                      </div>
                      <input
                        aria-autocomplete="list"
                        class="css-62g3xt-dummyInput"
                        id="react-select-3-input"
                        readonly=""
                        tabindex="0"
                        value=""
                      />
                    </div>
                    <div
                      class="sibyl-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="sibyl-select__indicator-separator css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="sibyl-select__indicator sibyl-select__dropdown-indicator css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-6q0nyr-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </li>
              <li>
                <div
                  class="sibyl-select css-2b097c-container"
                >
                  <div
                    class="sibyl-select__control css-yk16xz-control"
                  >
                    <div
                      class="sibyl-select__value-container css-g1d714-ValueContainer"
                    >
                      <div
                        class="sibyl-select__placeholder css-1wa3eu0-placeholder"
                      >
                        Contribution
                      </div>
                      <input
                        aria-autocomplete="list"
                        class="css-62g3xt-dummyInput"
                        id="react-select-4-input"
                        readonly=""
                        tabindex="0"
                        value=""
                      />
                    </div>
                    <div
                      class="sibyl-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="sibyl-select__indicator-separator css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="sibyl-select__indicator sibyl-select__dropdown-indicator css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-6q0nyr-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </li>
            </ul>
          </header>
          <div
            class="sticky-wrapper scroll-style"
            style="max-height:600px"
          >
            <table
              class="dash-table sticky-header"
            >
              <thead>
                <tr>
                  <th
                    class="align-center"
                  >
                    Category
                  </th>
                  <th>
                    Feature
                  </th>
                  <th
                    class="align-right"
                  >
                    Value
                  </th>
                  <th
                    class="align-center"
                    width="15%"
                  >
                    <ul
                      class="sort"
                    >
                      <li>
                        Contribution
                      </li>
                      <li>
                        <button
                          type="button"
                        >
                          <svg
                            height="18"
                            viewBox="0 0 10 18"
                            width="10"
                          >
                            <path
                              d="M4.99965 2.83021L7.45965 5.29021C7.84965 5.68021 8.47965 5.68021 8.86965 5.29021C9.25965 4.90021 9.25965 4.27022 8.86965 3.88021L5.69965 0.700215C5.30965 0.310215 4.67965 0.310215 4.28965 0.700215L1.11965 3.88021C0.729648 4.27022 0.729648 4.90021 1.11965 5.29021C1.50965 5.68021 2.13965 5.68021 2.52965 5.29021L4.99965 2.83021ZM4.99965 15.1702L2.53965 12.7102C2.14965 12.3202 1.51965 12.3202 1.12965 12.7102C0.739649 13.1002 0.739649 13.7302 1.12965 14.1202L4.29965 17.3002C4.68965 17.6902 5.31965 17.6902 5.70965 17.3002L8.87965 14.1302C9.26965 13.7402 9.26965 13.1102 8.87965 12.7202C8.48965 12.3302 7.85965 12.3302 7.46965 12.7202L4.99965 15.1702Z"
                              fill="#828282"
                            />
                          </svg>
                        </button>
                      </li>
                    </ul>
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td
                    class="align-center"
                  >
                    <i
                      class="bullet red"
                    />
                  </td>
                  <td>
                    child in focus was receiving County Diversion (CD) benefits at time of referral.
                  </td>
                  <td
                    class="align-right"
                  >
                    False
                  </td>
                  <td
                    class="align-center"
                    width="145"
                  >
                    <svg
                      class="bidi-progress-bar"
                      height="18"
                      width="110"
                    >
                      <rect
                        fill="rgba(189, 189, 189, 0.5)"
                        height="8"
                        rx="4"
                        width="110"
                        y="4"
                      />
                      <path
                        class="bar-negative"
                        d="M 55, 4
            h  0
            a  0,  0 0 0 1 0, 0
            v  8
            a  0, 0 0 0 1 -0, 0
            h  0
            z"
                      />
                      <rect
                        class="separator"
                        fill="#BDBDBD"
                        height="8"
                        rx="1"
                        ry="1"
                        width="2"
                        x="55"
                      />
                    </svg>
                  </td>
                </tr>
                <tr>
                  <td
                    class="align-center"
                  >
                    <i
                      class="bullet red"
                    />
                  </td>
                  <td>
                    child in focus received County Diversion (CD) benefits in the 1 year prior to referral, excl. clients who are actively receiving the service.
                  </td>
                  <td
                    class="align-right"
                  >
                    False
                  </td>
                  <td
                    class="align-center"
                    width="145"
                  >
                    <svg
                      class="bidi-progress-bar"
                      height="18"
                      width="110"
                    >
                      <rect
                        fill="rgba(189, 189, 189, 0.5)"
                        height="8"
                        rx="4"
                        width="110"
                        y="4"
                      />
                      <path
                        class="bar-negative"
                        d="M 55, 4
            h  0
            a  0,  0 0 0 1 0, 0
            v  8
            a  0, 0 0 0 1 -0, 0
            h  0
            z"
                      />
                      <rect
                        class="separator"
                        fill="#BDBDBD"
                        height="8"
                        rx="1"
                        ry="1"
                        width="2"
                        x="55"
                      />
                    </svg>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
